### PR TITLE
Disable S3 Select pushdown when the query does not filter data

### PR DESCRIPTION
**Pull request by dnanuti**
S3 Select pushdown improves queries performance by reducing the data on the wire and it is the most effective when the query pushed down to Select is filtering out a significant portion of the data. This commit disables Select pushdown when the query does not have a predicate or projection. In these cases using GET is both cheaper and faster.

Test plan - (Please fill in how you tested your changes)
Unit tests

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==
Hive Changes
* Disable S3 Select pushdown when the query does not have a predicate or projection
```

### CODE REVIEW

**Comment by dnanuti** [comment]
@pettyjamesm please have a look when you find the time

File: presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java
Comment (pettyjamesm) [inline presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java:130]: Would we still want to have S3 select enabled if all columns are selected but the schema types don't align? (would this be a "cast" situation?) It seems like regardless of the types involved, selecting all columns won't reduce data over the wire.

File: presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java
Comment (pettyjamesm) [inline presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java:143]: Seems like this could be `ImmutableSet.copyOf(columnNameProperty.split(columnNameDelimiiter))`

Comment (dnanuti) [reply presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectRecordCursorProvider.java]: Thanks! Nice catch!

**Review by pettyjamesm — APPROVED** [review]
LGTM, @rschlussel - anyone else you know of that should take a look before merging?

**Pull request merged**
